Extensions/Today/TodayViewController.swift

```diff
@@ -163,16 +209,128 @@ class TodayViewController: UIViewController, NCWidgetProviding {
             openContainingApp("firefox://?url=\(encodedString)")
         }
     }
+}
 
-    func updateOpenClipboardButton() {
-        log.info("Checking clipboard")
-        if let string = UIPasteboard.generalPasteboard().string,
-            _ = NSURL(string: string) {
-                log.info("We have a URL!")
-                openURLFromClipboardView.hidden = false
-        } else {
-            openURLFromClipboardView.hidden = true
+extension UIButton {
+    func setBackgroundColor(color: UIColor, forState state: UIControlState) {
+        let colorView = UIView(frame: CGRectMake(0, 0, 1, 1))
+        colorView.backgroundColor = color
+
+        UIGraphicsBeginImageContext(colorView.bounds.size)
+        if let context = UIGraphicsGetCurrentContext() {
+            colorView.layer.renderInContext(context)
+        }
+        let colorImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+
+        self.setBackgroundImage(colorImage, forState: state)
+    }
+}
+
+class ImageButtonWithLabel: UIView {
+
+    lazy var button = UIButton()
+
+    lazy var label = UILabel()
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    init() {
+        super.init(frame: CGRectZero)
+        performLayout()
+    }
+
+    func performLayout() {
+        addSubview(button)
+        addSubview(label)
+
+        button.snp_makeConstraints { make in
+            make.top.equalTo(self)
+            make.left.equalTo(self)
+            make.centerX.equalTo(self)
+        }
+
+        snp_makeConstraints { make in
+            make.width.equalTo(label)
+            make.height.equalTo(button)
+        }
+
+        label.snp_makeConstraints { make in
+            make.centerX.equalTo(button.snp_centerX)
+            make.centerY.equalTo(button.snp_centerY).offset(39)
+        }
+    }
+
+    func addTarget(target: AnyObject?, action: Selector, forControlEvents events: UIControlEvents) {
+        button.addTarget(target, action: action, forControlEvents: events)
+    }
+
+    override func hitTest(point: CGPoint, withEvent event: UIEvent?) -> UIView? {
```

Comment:
Not sure why you are overiding `hitTest` on the UIView. As long as `userInteractionEnabled = true` on the view it should work just fine.